std/node/_fs/_fs_dir.ts

```diff
@@ -29,7 +29,10 @@ export default class Dir {
     return new Promise(async (resolve, reject) => {
       try {
         if (this.initializationOfDirectoryFilesIsRequired()) {
-          const denoFiles: Deno.FileInfo[] = await Deno.readdir(this.path);
+          const denoFiles: Deno.DirEntry[] = [];
```

Comment:
`read()` and `readSync()` (for node polyfill) should be reworked.  The node polyfill current implementations had to mimic an iterator of directory entries by taking the array returned by the old `Deno.readdir()` and popping items from the array one at a time.  Now that Deno's readdir lines up with Node's readdir, this should be much simpler code to implement.  Happy for this to be done in a separate PR, as what is here should work, though it's a bit daft to take an iterator, create an array from it and then iterate over the array.

Reply:
> it's a bit daft to take an iterator, create an array from it and then iterate over the array.

And if you count how `Deno.readdir()` is implemented at the moment, we get the original iterator by getting an array from Rust and wrapping it in an iterator :)